test/rubygems/test_gem_commands_sources_command.rb

```diff
@@ -283,6 +316,39 @@ def test_execute_add_http_rubygems_org
     assert_empty @ui.error
   end
 
+  def test_execute_add_http_rubygems_org_forced
+    rubygems_org = "http://rubygems.org"
+
+    spec_fetcher do |fetcher|
+      fetcher.spec("a", 1)
+    end
+
+    specs = Gem::Specification.map do |spec|
+      [spec.name, spec.version, spec.original_platform]
+    end
+
+    specs_dump_gz = StringIO.new
+    Zlib::GzipWriter.wrap(specs_dump_gz) do |io|
+      Marshal.dump(specs, io)
+    end
+
+    @fetcher.data["#{rubygems_org}/specs.#{@marshal_version}.gz"] = specs_dump_gz.string
+    @cmd.handle_options %W[--force --add #{rubygems_org}]
+    ui = Gem::MockGemUi.new("y")
```

Comment:
Same here about the UI, but I wonder if we need this test at all since this case will be fixed by https://github.com/rubygems/rubygems/pull/3951, rright?

Reply:
Oh, I see, thanks!